package.json

```diff
@@ -17,6 +17,7 @@
   ],
   "scripts": {
     "dev": "lerna run --parallel --scope @firebase/* --scope firebase dev",
+    "precommit": "lint-staged",
```

Comment:
Oh boy...  I don't know if I trust prettier enough to let it silently modify my commits.  Then I can't tell what it did and I might not notice problems.  Today I always run "yarn prepush" manually, check what it did, and then if I'm satisfied, I'll clean up the git history via "git rebase -i master" or whatever.

Reply:
Using the staged files makes it really easy to figure out what files to re-format. A while ago, I tried something similar to what you suggested in https://github.com/firebase/firebase-js-sdk/issues/531 (diff against origin & upstream) but wasn't able to get it to work properly in between my other work. 

I personally don't like the Prettier commits but I also don't want to take them away from you (since I can just rebase if need be). Maybe there is some way we can still stage them in a separate commit (by not passing `--write`, or not doing `git add`?). Do you have suggestions for this, @jshcrowthe?

I do want to ask though if it is even possible to disagree with prettier. If you manually re-format prettier's changes, doesn't it just revert these changes on the next run?

Reply:
I don't know what this would entail, but we could do a `post-commit` hook and just try and format the last diff? I dunno what it would require/look like, but it seems doable.

Reply:
I sometimes choose to wrap lines differently (or restructure code to avoid wrapping, etc.).  And I just like to know what prettier is doing. We've run into cases where it just does wonky reformatting (either due to bugs or poor design) and I like to notice those when they happen.